### PR TITLE
Re-implement decorate stacktrace

### PR DESCRIPTION
**Pull request by phillip-kruger**
The decorate PR (#37034) that was merged a few days ago broke the log help :

[x] - Open last exception (or project) in IDE

This PR re implements the decorate feature in a safer way. (The old implementation clears the stacktraceitems array).

The previous PR (that is not in a release yet) changed the Throwable by setting the stacktrace to an empty array, and override the toString. This broke anyone else downstream that is looking at the stacktrace elements.

This PR does not change the stacktrace item array, but rather change the message (for console only) just before we write to console. For the error page we just include the decoration on the page, before the exception is printed.

**Now in the log:**
error_console

And then pressing x will open the correct file on the correct place in your IDE

**And now in the error page:**
error_page

And (this is new) clicking on the link will open the correct file on the correct place in your IDE

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/logging/LoggingDecorateBuildItem.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.CompositeIndex;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+
+public final class LoggingDecorateBuildItem extends SimpleBuildItem {
```
Comment (geoand) [inline core/deployment/src/main/java/io/quarkus/deployment/logging/LoggingDecorateBuildItem.java:13]: Can we have some Javadoc on this?

**Review by geoand — COMMENTED** [review]
> This PR re implements the decorate feature in a safer way. (The old implementation clears the stacktraceitems array.

Can we also have a description here of what the old way vs the new way is for posterity?

**Comment by phillip-kruger** [comment]
@geoand I updated the description and added the Javadoc. Thanks

**Review by geoand — APPROVED** [review]
Excellent!

**Pull request merged**